### DISCUSSION TITLE
Postgres Written Data Limit always unreasonably high

### DISCUSSION THREAD

**Discussion by unoflavora** [vercel/community#4558] [Bug Report]
### Summary

Helloso i do two queries which is inserting user data (TWO users), and then the written data in my Postgres billing becomes 74MB, this is doesn't make any sense. Please give me more information on why this is happening and what I can do so this is never happening again.

<img width="614" alt="image" src="https://github.com/vercel/community/assets/72590089/f6c8684d-6af6-420c-aee4-bebac5bab9f2">

### Steps to Reproduce

- create new database
- connect it to the app
- see that the value of written data in DB is 0 mb
- add one or two rows of data
- see that now the written data become exorbitantly high

**Comment by NeMaksym** [comment]
Same
I have 5 tables in DB. Used it only for local development and did about 100 records max.
```
Written data: 224.23 MB / 256 MB (88%) 
Storage: 45.99 MB / 256 MB (18%)
```

**Comment by InternetJohnny** [comment]
This happened to me without even writing records to the database.
Here is my issue ([#4815](https://github.com/orgs/vercel/discussions/4815)).

**Comment by bmmathe** [comment]
Same for me.  I created a db yesterday and wrote about 60 small records to it and now my storage is over 150MB even though my transfer is ~300kB.  How is it possible that the db is 33MB base but my storage this month (starting yesterday) is over 150MB?  Seeing this definitely makes me not want to consider using this in a Pro plan.

**Reply by InternetJohnny** [reply]
Definitely think going with another option is better since it is just not reasonable that a few transactions/migrations would use 300MB.
I went with Planetscale. Many others recommend it, Supabase or Neon (what Vercel Postgres is using, they just add a large markup on top).
You can see some comparisons here between what Vercel offers and the services they use, it really is crazy.
https://service-markup.vercel.app/

**Comment by unoflavora** [comment]
> Definitely think going with another option is better since it is just not reasonable that a few transactions/migrations would use 300MB. I went with Planetscale. Many others recommend it, Supabase or Neon (what Vercel Postgres is using, they just add a large markup on top). You can see some comparisons here between what Vercel offers and the services they use, it really is crazy. https://service-markup.vercel.app

Wow, that's a big markup. I always know that they have some price markups but I always think that is justified because they making good work on Next.js. It's just that if they cannot fix a major bug like this, I cannot see that the justification is valid anymore, moreover after seeing that big markup.

**Reply by leerob** [reply]
This information isn't accurate - prices have changed https://vercel.com/pricing

**Reply by bmmathe** [reply]
For the me the problem isn't a flat markup, it's inaccurate reporting.  There is no possible way I have used 169MB of database storage with 60 small records.  The database is 33MB plus ~100kb of data = 169MB?

**Reply by bmmathe** [reply]
@leerob Any information on why this bug still exists.  There are several other threads about the same issue and the response was "it is a bug and we fixed it" however it's not fixed.  I am now locked out of using my plan even though I only have < 100 records in it (very small amount of data) yet my project says I have transferred over 300MB.

**Comment by cagkanacarbay** [comment]
I have the same issue. Uploaded maybe 40 short records in total. already at the 256mb limit.

**Reply by cagkanacarbay** [reply]
I contacted the team about this issue. They're telling me this may be caused by shadow database made by prisma.: https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database

Everytime you run `prisma migrate dev` a new database is created, which could be the reason behind this.

**Reply by isaacsultan** [reply]
Thanks @cagkanacarbay - how did you end up resolving this?

**Reply by cagkanacarbay** [reply]
I decided not to use vercel 😂

**Comment by onurhan1337** [comment]
Hey 🙌  I have an issue like you. Could you find any way to fix it?

**Comment by ivanvolti** [comment]
I'm facing the same issue. The day before yesterday, I made a PostgreSQL database on Vercel and completed just 4 migrations during the development with Prisma. Today, Vercel sent me a message saying: "Your account ... has reached its full free limit for data writing, which is 256.00 MB." This is confusing because I haven't even finished my work to get it ready for actual use.
![CleanShot 2024-01-14 at 13 48 15@2x](https://github.com/vercel/community/assets/1476221/2f550131-67ea-4f82-a39f-63ccdd5771e2)

**Comment by kimuradev** [comment]
Any updates?
